dna/modify.go

```diff
@@ -133,6 +150,22 @@ func Insert(seq []Base, insPos int64, insSeq []Base) []Base {
 	return append(seq[:insPos], append(insSeq, seq[insPos:]...)...)
 }
 
+// InsertStable is identical to Insert, but conserves the
+// underlying memory of the input destSeq slice if possible
+func InsertStable(destSeq *[]Base, insPos int, insSeq []Base) error {
+	if insPos < 0 || insPos > len(*destSeq) {
+		return ErrInvalidPosition
+	}
+	if len(*destSeq)+len(insSeq) > cap(*destSeq) {
+		*destSeq = Insert(*destSeq, int64(insPos), insSeq)
+		return ErrExceedCapNewSliceCreated
+	}
+	*destSeq = (*destSeq)[:len(*destSeq)+len(insSeq)]                                    // extend the LEN of the slice to include the inserted bases
+	copy((*destSeq)[insPos+len(insSeq):cap(*destSeq)], (*destSeq)[insPos:cap(*destSeq)]) // shift bases in slice by len(insSeq) at the position of the insertion
```

Comment:
I'm trying to wrap my head around how copy() works. Can you point me in the direction of some documentation please?

Reply:
See https://yourbasic.org/golang/copy-explained/

Basically copy just copies the content from one slice into another slice. Here i use copy to "shift" the elements in the slice to make room for more elements to be added. Its not a very intuitive use of the copy function, but it is very handy.

Reply:
That helped a lot, thanks Dan! I think I was getting caught up on the fact that the insert was beginning at the end point of the insertion but I realize it doesn't matter since the total number of slices will increase which is the ultimate goal.